go/pkg/client/cas_test.go

```diff
@@ -250,22 +250,34 @@ func TestRead(t *testing.T) {
 			}
 
 			if tc.offset == 0 && tc.limit == 0 {
-				got, err := c.ReadBlob(ctx, digest.NewFromBlob(want))
+				got, bMoved, err := c.ReadBlob(ctx, digest.NewFromBlob(want))
 				if err != nil {
 					t.Errorf("c.ReadBlob(ctx, digest) gave error %s, want nil", err)
 				}
 				if !bytes.Equal(want, got) {
 					t.Errorf("c.ReadBlob(ctx, digest) gave diff: want %v, got %v", want, got)
 				}
+				if tc.compress && bMoved == int64(len(got)) && len(tc.fake.Blob) != 0 {
```

Comment:
It's deterministic given some parameters, eg compression speed used. IIUC, upgrading the underlying compression lib or just tweaking the settings we use for it could arguably change the compressed value to something that'd still be valid zstandard. So I think using concrete values to check against would be more trouble than it's worth. Wdyt?

Reply:
I"m hoping we won't do this often and even then we wouldn't be changing the defaults?

Reply:
So I think my options here to use the actual expected values for `got` and `bMoved` are:
1) Have it explicitly in the test cases. However, I think it's much simpler and cleaner to just run all test cases for compression both on and off.
2) I'd still have to add a `if tc.compress` to check against decide to check against the compressed or non-compressed real values.

So I went in a slightly different path here: as per Ola's suggestions to always return `MovedBytesMetadata`, I now do already have the logical and real bytes, so the checks are slightly simpler (the first one is for both compressed and non-compressed cases, the second one is only for compressed cases).

Wdyt?